### PR TITLE
Feature/show media shows cw

### LINKED ISSUE #72: Revealing sensitive images should also reveal CWed text
Currently: if a CWed post has images, you see the CW and blurred-out versions of the images. If you click on the images, they appear - but the toot body does not. It is easy to forget that there *is* text in the toot body. (But there always is because if there weren't the post would have had its CW status removed and the CW text would become the post body.)

I suggest: clicking on a sensitive image in a CWed toot reveals both image and body text. There are three states: CW and blurred image, CW and body text and blurred image, CW and body text and revealed image. (The fourth state, CW, hidden body, revealed image, is what I am suggesting removing.)

I also suggest: Clicking on "show more" does *not* reveal the image; this is just a feeling that there will be cases where users are comfortable with the text but unsure about viewing the image - especially without seeing the text first. I think images are more likely to have a visceral impact so we should be conservative about showing them.

### PR DESCRIPTION
**Pull request by robinmaypanpan**
Fixes #72

### CODE REVIEW

File: app/javascript/mastodon/components/media_gallery.js [hunk 1]
```diff
@@ -244,31 +244,32 @@ class MediaGallery extends React.PureComponent {
...
-
-  handleOpen = () => {
-    this.setState({ visible: !this.state.visible });
-  }
-
   handleClick = (index) => {
     this.props.onOpenMedia(this.props.media, index);
   }
 
+  handleMediaClick = (index) => {
```
Comment (aarchiba) [inline app/javascript/mastodon/components/media_gallery.js:264]: Do you have a choice about this name? It sounds like this should handle any click on the media not just on the visibility icon.

File: BC_CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,9 @@ Changelog
 All changes to Beach City beyond the vanilla Mastodon code will be documented here.
 ## [1.7.2] - 2019-06-??
 
+### Changed
+- Opening sensitive image will open CWs
```
Comment (aarchiba) [inline BC_CHANGELOG.md:8]: Will closing the CW also close the media?

**Review by aarchiba — APPROVED** [review]
One confusing name, and one question about the converse action, otherwise it looks solid.

File: app/javascript/mastodon/components/media_gallery.js (hunk 1 above)
Comment (robinmaypanpan) [reply app/javascript/mastodon/components/media_gallery.js:264]: That is what it does though!  It also follows the pattern from the handleClick function above.

File: BC_CHANGELOG.md (hunk 2 above)
Comment (robinmaypanpan) [reply BC_CHANGELOG.md:8]: No.

File: app/javascript/mastodon/components/media_gallery.js (hunk 1 above)
Comment (aarchiba) [reply app/javascript/mastodon/components/media_gallery.js:264]: But this function seems to toggle the hidden state? We don't want a click on un-hidden media to hide it again, we want that to pop up the viewer.

**Review by aarchiba — COMMENTED** [review]
Potential bug?

**Pull request merged**